### PR TITLE
HPCC-16085 MySQL plugin connection cache needs to retry on failure

### PR DESCRIPTION
**Pull request by richardkchapman**
When reusing a cached MySQL connection, we need to retry on failures in case
the MySQL server has restarted.

Signed-off-by: Richard Chapman rchapman@hpccsystems.com

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review. Mostly this is code moving around...

File: plugins/mysql/mysqlembed.cpp [hunk 1]
```diff
@@ -1505,12 +1520,24 @@ class MySQLEmbedFunctionContext : public CInterfaceOf<IEmbedFunctionContext>
     virtual void compileEmbeddedScript(size32_t chars, const char *script)
     {
         size32_t len = rtlUtf8Size(chars, script);
-        Owned<MySQLStatement> stmt  = new MySQLStatement(mysql_stmt_init(*conn));
-        if (!*stmt)
-            fail("failed to create statement");
-        if (mysql_stmt_prepare(*stmt, script, len))
-            fail(mysql_stmt_error(*stmt));
-        stmtInfo.setown(new MySQLPreparedStatement(conn, stmt));
+        loop
```
Comment (richardkchapman) [inline plugins/mysql/mysqlembed.cpp]: This is slightly odd code - perhaps a goto would be clearer?

Comment (ghalliday) [inline plugins/mysql/mysqlembed.cpp]: Or you could reorder the tests.

**Comment by ghalliday** [comment]
minor: unrelated to this pull, but the code

                Owned<MySQLConnection> cacheEntry = new MySQLConnection(*this);

Is really a good example of where a move constructor should be used (and cacheOptions/conn should be cleared within that move constructor)

File: plugins/mysql/mysqlembed.cpp (hunk 1 above)
Comment (richardkchapman) [inline plugins/mysql/mysqlembed.cpp]: Not sure what you are suggesting

**Comment by richardkchapman** [comment]
I'll learn to use move constructors one day...

**Pull request merged**